src/cascade/input_data/configuration/construct_country.py

```diff
@@ -217,3 +226,187 @@ def convert_gbd_ids_to_dismod_values(with_ids_df, age_groups_df):
+    This allows checks like: 9.5 in age_interval (yes), 20.25 in age_interval (no),
+    0.667 in age_interval (no), 95 in age_interval (no), 50.1 in age_interval (yes)
+
+    Returns:
+        Interval: of ages in the covariates data frame
+    """
+    age_interval = it.empty()
+
+    age_groups = set(zip(covariates["age_lower"], covariates["age_upper"]))
+
+    for age in age_groups:
+        age_interval = age_interval | it.closed(age[0], age[1])
+
+    return age_interval
+
+
+def assign_interpolated_covariate_values(measurements, sex, covariates):
+    """
+    Compute a column of covariate values to assign to the measurements.
+
+    Args:
+        measurements (pd.DataFrame):
+            Columns include ``age_lower``, ``age_upper``, ``time_lower``,
+            ``time_upper``. All others are ignored.
+        sex (pd.Series): The sex covariate as [-0.5, 0, 0.5], corresponding to the measurements.
+        covariates (pd.DataFrame):
+            Columns include ``age_lower``, ``age_upper``, ``time_lower``,
+            ``time_upper``, ``sex``, and ``value``.
+    Returns:
+        pd.Series: One row for every row in the measurements.
+    """
+
+    # is the covariate by_age, does it have multiple years?
+    covar_at_dims = compute_covariate_age_time_dimensions(covariates)
+
+    # identify the overall interval for the covariate ages, could have middle gaps
+    covar_age_interval = compute_covariate_age_interval(covariates)
+
+    # find a matching covariate value for each measurement
+    covariate_column = compute_interpolated_covariate_values_by_sex(
+        measurements, covariates, covar_at_dims, covar_age_interval)
+
+    return covariate_column
+
+
+def get_measurement_data_by_sex(measurements):
+    """Split the measurement data by sex values found in the measurements.
+    Args:
+        measurements (pandas.DataFrame): data for a specific measurement
+    Returns:
+        dict: possible sex keys (-0.5, 0, 0.5) and measurement data as values
+    """
+    measurements_by_sex = {}
+
+    for sex in (FEMALE, MALE, BOTH):
+
+        measurements_sex = measurements[measurements["x_sex"] == sex]
+
+        if not measurements_sex.empty:
+            measurements_by_sex[sex] = measurements_sex
+
+    return measurements_by_sex
+
+
+def compute_interpolated_covariate_values_by_sex(
+        measurements, covariates, covar_at_dims, covar_age_interval):
+    """
+    Use the measurements data by sex as input to the corresponding interpolator
+    to assign a covariate value to the measurement.
+
+    Returns:
+        pd.Series: One row for every row in the measurements.
+    """
+
+    covariates["avg_age"] = covariates[["age_lower", "age_upper"]].mean(axis=1)
+    covariates["avg_time"] = covariates[["time_lower", "time_upper"]].mean(axis=1)
+
+    covariates_by_sex = get_covariate_data_by_sex(covariates)
+
+    measurements["avg_age"] = measurements[["age_lower", "age_upper"]].mean(axis=1)
```

Comment:
Same question as above. There's some code for this in EMR: https://github.com/ihmeuw/cascade/blob/develop/src/cascade/input_data/emr.py#L40

Also this for the relevant policy: `model_context.policies["use_weighted_age_group_midpoints"]`

We should probably extract this logic out into a shared utility since this as well as EMR need it and probably other places too.